talib/_abstract.pxi

```diff
@@ -348,8 +356,43 @@ class Function(object):
         This is a shortcut to the outputs property that also allows setting
         the input_arrays dict and function parameters.
         """
+        # do not cache ta-func parameters passed to __call__
+        opt_input_values = [(param_name, self.__opt_inputs[param_name]['value'])
+                            for param_name in self.__opt_inputs.keys()]
+
+        # allow calling with same signature as talib.func module functions
+        args = list(args)
+        input_arrays = {}
+        input_price_series_names = self.__input_price_series_names()
+        if args and not isinstance(args[0], __INPUT_ARRAYS_TYPES):
+            for i, arg in enumerate(args):
+                if not isinstance(arg, __ARRAY_TYPES):
+                    break
+
+                try:
+                    input_arrays[input_price_series_names[i]] = arg
+                except IndexError:
+                    msg = 'Too many price arguments: expected %d (%s)' % (
+                        len(input_price_series_names),
+                        ', '.join(input_price_series_names))
+                    raise TypeError(msg)
+
+        if len(input_arrays) == len(input_price_series_names):
+            self.set_input_arrays(input_arrays)
+            args = args[len(input_arrays):]
+        elif len(input_arrays):
```

Comment:
If no arrays are passed, maybe it should still error if some price arguments are expected?

Instead I get:

```
>>> talib.abstract.Function("MAMA")()
Traceback (most recent call last):
  File "/Users/jbenedik/Dev/ta-lib/talib/__init__.py", line 20, in wrapper
    for arg in chain(args, kwargs.values())
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "talib/_abstract.pxi", line 390, in talib._ta_lib.Function.__call__
    self.__call_function()
  File "talib/_abstract.pxi", line 426, in talib._ta_lib.Function.__call_function
    results = self.func_object(*args)
  File "/Users/jbenedik/Dev/ta-lib/talib/__init__.py", line 24, in wrapper
    return func(*args, **kwargs)
TypeError: Argument 'real' has incorrect type (expected numpy.ndarray, got NoneType)
```

Also, I'm not so keen on the nested exceptions with our pandas function wrapper... need to clean that up maybe.

Reply:
Okay, fixed the nested exception "during handling of ..." in 3e29a24100fc6d534b2f60a9be2bb47b96c3ebd1.